api4/system.go

```diff
@@ -545,3 +548,70 @@ func getServerBusyExpires(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 	w.Write([]byte(c.App.Srv().Busy.ToJson()))
 }
+
+func upgradeToEnterprise(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_SYSTEM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+		return
+	}
+
+	auditRec := c.MakeAuditRecord("upgradeToEnterprise", audit.Fail)
+	defer c.LogAuditRec(auditRec)
```

Comment:
Is there a reason for auditing the request after the permission check?

Reply:
I think was based on existing code, and I've seen that there are `MakeAuditRecord` calls that are done before the permissions check and other after. I asked in the developers-server channel about it, but for now I going to move it before the permissions check.

Reply:
Done